zerver/lib/users.py

```diff
@@ -511,6 +515,45 @@ def check_user_can_access_all_users(acting_user: Optional[UserProfile]) -> bool:
     return False
 
 
+def check_can_access_user(
+    target_user: UserProfile, user_profile: Optional[UserProfile] = None
+) -> bool:
+    if check_user_can_access_all_users(user_profile) or not user_access_restricted_in_realm(
+        target_user
+    ):
+        return True
+
+    assert user_profile is not None
+
+    if target_user.id == user_profile.id:
+        return True
```

Comment:
We probably will need a bonus special case for system bots here, so that one can mute them, right?

(I can't remember if we made that work elsewhere, but this seems logical).

Reply:
The system bots will also have `is_bot` field set to `True` so `user_access_restricted_in_realm` should hanlde that I guess.